src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -252,7 +253,53 @@ public static string LocalSlotName
 
             this.logger.LogInformation(
                 "{instanceId}: Function '{functionName} ({functionType})', version '{version}' received a '{eventName}' event. State: {state}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {extensionVersion}. SequenceNumber: {sequenceNumber}.",
-                instanceId, functionName, functionType, version, eventName, FunctionState.ExternalEventRaised, hubName, LocalAppName, LocalSlotName, ExtensionVersion, this.sequenceNumber++);
+                instanceId, functionName, functionType, version, eventName, FunctionState.ExternalEventRaised, hubName,
+                LocalAppName, LocalSlotName, ExtensionVersion, this.sequenceNumber++);
+        }
+
+        public void SendMessageFailed(
```

Comment:
Let's give this a more specific method name, like `EventGridNotificationFailed`.